File: Codechef/NotATriangle.cpp
#include <iostream>
#include <algorithm>


using namespace std;

int main()
{
	int t,arr[2000];
	cin>>t;
	while(t != 0)
	{
		int count = 0;
		for(int i=0;i<t;i++)
			cin>>arr[i];
		sort(arr , arr+t);

		for(int i=0;i<t-2;i++)
		{
			int k = i+2;

			for (int j = i+1; j < t-1; ++j)
			{
				while(arr[i]+arr[j] >= arr[k] && k<t)
				{
					k++;
				}	
				count += t-k;
			}
		}

		cout<<count<<endl;
		cin>>t;

	}
	return 0;
}

File: Codechef/notATriangle.py
import math

def istriangle(a,b,c):
	if( a+b < c or a+c < b or b+c < a):
		return True

	return False

def main():

	n = raw_input()
	n = int(n)
	
	while n != 0:
		count = 0
		arr = raw_input()
		arr = arr.split()
		arr = map(int , arr)
		arr.sort()
		#print arr

		if n >= 3 :	
			for i in range (0,n-2) :
				for j in range(i+1,n-1):
					for k in range(j+1,n):
						if(istriangle(arr[i],arr[j],arr[k])):
							count = count + n-k
							break
						
			print count
		
		else :
			print 0

		n = raw_input()
		n = int(n)

if __name__ == '__main__':
	main()